lib/errors/index.js

```diff
@@ -217,20 +217,96 @@ exports.UnknownConstraintError = UnknownConstraintError;
  * Validation Error Item
  * Instances of this class are included in the `ValidationError.errors` property.
  *
- * @param {string} message An error message
- * @param {string} type The type of the validation error
- * @param {string} path The field that triggered the validation error
+ * @param {String} message  - An error message
+ * @param {String} type     - The type of the validation error
+ * @param {String} path     - The field that triggered the validation error
+ * @param {String} value    - The value that generated the error
+ * @param {Object} inst     - optional, the DAO instance that caused the validation error
+ * @param {Object} vKey     - optional, a validation "key", used for identification
+ * @param {String} fnName   - optional, property name of the BUILT-IN validator function that caused the validation error (e.g. "in" or "len"), if applicable
+ * @param {String} fnArgs   - optional, parameters used with the BUILT-IN validator function, if applicable
+ *
  * @param {string} value The value that generated the error
  */
 class ValidationErrorItem {
-  constructor(message, type, path, value) {
+  constructor(message, type, path, value, inst, vKey, fnName, fnArgs) {
     this.message = message || '';
-    this.type = type || null;
+    this.type = null;
     this.path = path || null;
     this.value = value !== undefined ? value : null;
+
+    this.instance       = inst || null;
+
+    this.validatorKey   = vKey || null;
+    this.validatorName  = fnName || null;
+    this.validatorArgs  = fnArgs || [];
+
+    if (type) {
+      if (ValidationErrorItem.USE_OLD_TYPES || ValidationErrorItem.TYPES[ type ]) {
+        this.type = type;
+      } else {
+        const lctype = (type + '').toLowerCase().trim();
+        const rtype  = ValidationErrorItem.TypeStringMap[ lctype ]; // "real type" ... but also https://en.wikipedia.org/wiki/R-Type :-D
+
+        if (rtype && ValidationErrorItem.TYPES[ rtype ]) {
+          this.type           = rtype;
+          this.originalType   = type; // expose the value of "type" as it was in previous version (just in case someone is reliant on it?)
```

Comment:
the reasoning for breaking BC is in the description of the PR - TL;DR = current property contains arbitrary strings which we should not encourage people to rely on.

if you don't agree with the reason please let me know what it is you don't agree with exactly